packages/emotion/native/Css.ml

```diff
@@ -298,40 +300,67 @@ let keyframes_to_string keyframes =
   in
   keyframes |> List.map pp_keyframe |> String.concat ""
 
-let render_hash prefix hash styles =
+let render_hash hash styles =
   let is_label = function D ("label", value) -> Some value | _ -> None in
   match List.find_map is_label styles with
-  | None -> Printf.sprintf "%s-%s" prefix hash
-  | Some label -> Printf.sprintf "%s-%s-%s" prefix hash label
+  | None -> Printf.sprintf "%s" hash
+  | Some label -> Printf.sprintf "%s-%s" hash label
 
 let style (styles : rule list) =
   match styles with
   | [] -> ""
   | _ ->
+    let preffix = "css" in
     let hash = Murmur2.default (rules_to_string styles) in
-    let className = render_hash "css" hash styles in
-    Stylesheet.push instance (className, Classnames styles);
-    className
+    let className = render_hash hash styles in
+    Stylesheet.push instance (preffix, className, Classnames styles);
+    Printf.sprintf "%s-%s" preffix className
+
+let global (styles : rule list) =
+  match styles with
+  | [] -> ""
+  | _ ->
+    let hash = Murmur2.default (rules_to_string styles) in
+    Stylesheet.push instance ("", hash, Global styles);
+    hash
 
 let keyframes (keyframes : (int * rule list) list) =
   match keyframes with
   | [] -> ""
   | _ ->
+    let preffix = "animation" in
     let hash = Murmur2.default (keyframes_to_string keyframes) in
-    let animationName = Printf.sprintf "%s-%s" "animation" hash in
-    Stylesheet.push instance (animationName, Keyframes keyframes);
-    animationName
+    Stylesheet.push instance (preffix, hash, Keyframes keyframes);
+    Printf.sprintf "%s-%s" preffix hash
 
+(** Deprecated: Use get_style_rules instead*)
 let render_style_tag () =
   Stylesheet.get_all instance
   |> List.fold_left
-       (fun accumulator (hash, rules) ->
+       (fun accumulator (preffix, hash, rules) ->
+         let name = Printf.sprintf "%s-%s" preffix hash in
          match rules with
+         | Global rules ->
+           Printf.sprintf "%s %s" accumulator (rules_to_string rules)
          | Classnames rules ->
-           let rules = pp_rules hash rules |> String.trim in
+           let rules = pp_rules name rules |> String.trim in
            Printf.sprintf "%s %s" accumulator rules
          | Keyframes keyframes ->
-           let rules = pp_keyframes hash keyframes |> String.trim in
+           let rules = pp_keyframes name keyframes |> String.trim in
            Printf.sprintf "%s %s" accumulator rules)
        ""
   |> String.trim
+
+let get_string_style_rules = render_style_tag
+
+let get_string_style_hashes () =
+  Stylesheet.get_all instance
+  |> List.fold_left
+       (fun accumulator (_, hash, _) ->
+         Printf.sprintf "%s %s" accumulator hash |> String.trim)
+       ""
+
+let render_style_html_tag () =
```

Comment:
You moved away from the idea of having a component here?

Reply:
I forgot to add it.
I'm going to apply it like this:

```ocaml
let style_tag =
  React.createElement "style"
    [
      String ("data-emotion", "css " ^ get_string_style_hashes ());
      Bool ("data-s", true);
      DangerouslyInnerHtml (get_string_style_rules ());
    ]
    []
```

Works for you?

Reply:
@davesnx I was trying to create this style_tag to use like <Css.styleTag/>

```ocaml
let style_tag ?key:_ ()=
  React.createElement "style"
    [
      String ("data-emotion", "css " ^ get_string_style_hashes ());
      Bool ("data-s", true);
      DangerouslyInnerHtml (get_string_style_rules ());
    ]
    []
```

But I'm receiving this issue:

<img width="520" alt="image" src="https://github.com/davesnx/styled-ppx/assets/35539594/99c30995-b479-42d8-bc46-317b901d9e44">

A workaround for it is:

```ocaml
let style_tag ?key:_ ~children:_ ()=
  React.createElement "style"
    [
      String ("data-emotion", "css " ^ get_string_style_hashes ());
      Bool ("data-s", true);
      DangerouslyInnerHtml (get_string_style_rules ());
    ]
    []
```

But seems strange to have children that isn't used and even without breaking the calling when adding it:
`<Css.styleTag/> children </Css.styleTag>` It should have an error since it shouldn't receive a children value.

Should I use the workaround or do you have any idea how to handle it?

Reply:
The error message isn't the best, but it tells about a missing `children` argument, which should be there, so the function should be:

```
let style_tag ?key:_ () =
  React.createElement "style"
    [
      String ("data-emotion", "css " ^ get_string_style_hashes ());
      Bool ("data-s", true);
      DangerouslyInnerHtml (get_string_style_rules ());
    ]
    []
```

actually, not even key should be needed in this case, it doesn't hurt but isn't needed by any chance.

Reply:
I thought the same, but I guess it is the way JSX is being handled, when calling the make function from the component as <Component />, it expects those parameters. Does it make sense?

Reply:
Random idea: Do you have server-reason-react.react added as dependency?

Reply:
I'm testing it on e2e-melange which has this dune for debug:
```
(executable
 (name debug)
 (public_name e2e_melange_debug)
 (modules :standard \ index)
 (libraries ui_native server-reason-react.react server-reason-react.reactDom styled-ppx.emotion_native styled-ppx.css_native)
 (preprocess
  (pps styled-ppx)))
```